File: SellGUI-DynamicShop/src/main/java/me/aov/sellgui/dynamicshop/SellGUIDynamicShop.java
package me.aov.sellgui.dynamicshop;

import me.aov.sellgui.SellGUIMain;
import org.bukkit.plugin.java.JavaPlugin;

public final class SellGUIDynamicShop extends JavaPlugin {
    private DynamicShopPriceProvider priceProvider;

    @Override
    public void onEnable() {
        saveDefaultConfig();

        SellGUIMain sellGUI = SellGUIMain.getInstance();
        if (sellGUI == null || sellGUI.getSellGUIAPI() == null) {
            getLogger().severe("SellGUI API is not available. Disabling SellGUI-DynamicShop.");
            getServer().getPluginManager().disablePlugin(this);
            return;
        }

        priceProvider = new DynamicShopPriceProvider(this);
        sellGUI.getSellGUIAPI().registerPriceProvider(priceProvider);
        getLogger().info("Registered DynamicShop price provider with SellGUI.");
    }

    @Override
    public void onDisable() {
        SellGUIMain sellGUI = SellGUIMain.getInstance();
        if (sellGUI != null && sellGUI.getSellGUIAPI() != null && priceProvider != null) {
            sellGUI.getSellGUIAPI().unregisterPriceProvider(priceProvider);
        }
    }
}

File: SellGUI-DynamicShop/src/main/kotlin/me/aov/sellgui/dynamicshop/SellGUIDynamicShop.kt
package me.aov.sellgui.dynamicshop

import me.aov.sellgui.SellGUIMain
import org.bukkit.plugin.java.JavaPlugin

class SellGUIDynamicShop : JavaPlugin() {
    private var priceProvider: DynamicShopPriceProvider? = null

    override fun onEnable() {
        saveDefaultConfig()
        val sellGUI = SellGUIMain.getInstance()
        if (sellGUI == null || sellGUI.getSellGUIAPI() == null) {
            logger.severe("SellGUI API is not available. Disabling SellGUI-DynamicShop.")
            server.pluginManager.disablePlugin(this)
            return
        }
        priceProvider = DynamicShopPriceProvider(this)
        sellGUI.getSellGUIAPI().registerPriceProvider(priceProvider)
        logger.info("Registered DynamicShop price provider with SellGUI.")
    }

    override fun onDisable() {
        val sellGUI = SellGUIMain.getInstance()
        val provider = priceProvider
        if (sellGUI?.getSellGUIAPI() != null && provider != null) sellGUI.getSellGUIAPI().unregisterPriceProvider(provider)
    }
}
